packages/ui/src/machines/authMachine.ts

```diff
@@ -110,9 +112,33 @@ export const authMachine = createMachine<AuthContext, AuthEvent>(
       clearUser: assign({
         user: undefined,
       }),
-      setAuthConfig: assign({
-        config(_, event) {
-          return event.data.auth;
+      applyAmplifyConfig: assign({
+        config(context, event) {
+          const loginMechanisms = event.data.aws_cognito_login_mechanisms?.map(
+            (login) => {
+              switch (login) {
+                case 'PREFERRED_USERNAME':
+                case 'USERNAME':
+                  return 'username';
+
+                case 'EMAIL':
+                case 'PHONE_NUMBER':
+                case 'FACEBOOK':
+                case 'GOOGLE':
+                case 'AMAZON':
+                case 'APPLE':
+                  return login.toLowerCase();
+                default:
+                  console.warn(`Unknown login mechanism: ${login}`);
+              }
+            }
+          );
+
+          return {
+            loginMechanisms,
+            // Override config defaults with explicit values provided to state machine
+            ...context.config,
```

Comment:
^ Note this part. This may not work as expected until we test zero-config stuff.  For example, is `USERNAME` always present? Should defaulting `['username']` in the state machine happen _here_ rather than on init?

Should `context.config.login_mechanisms = getDefaultLoginMechanisms(Amplify.configure())` at run-time?

I don't know yet until the feature fully lands and we basically _remove_ all `loginMechanism` usage from our examples and re-test.

Reply:
imo, i think what you have is fine as-is for now, until we get the feature at hand. And once it lands, my understanding is that I expect all aws-exports to have the zero-config values if customer ran `amplify pull`, so I don't see much need for `getDefaultLoginMechanisms` on runtime.

Reply:
Btw, with the above implementation `loginMechanisms` will always be overwritten by `...context.config` because it's not deep merging. Is this is intended so that zero config doesn't take effect for now (and we'll flatten config object as suggested in above comment)?

Reply:
@wlee221 You're 100% correct.  The only thing we'll need to do is map the UPPERCASE values to our lowercase input equivalents.

But this logic isn't actually correct (yet). Fixing this in 08bfd89bde23986f648518c1918b7905024302d3 so that zero-config will "just work" 🙏